src/components/editableTable/GroupEditableTableRow.js

```diff
@@ -0,0 +1,266 @@
+import React, { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import TableRow from '@material-ui/core/TableRow';
+import TableCell from '@material-ui/core/TableCell';
+import EditIcon from '@material-ui/icons/Edit';
+import Save from '@material-ui/icons/Save';
+import Clear from '@material-ui/icons/Clear';
+import Delete from '@material-ui/icons/Delete';
+import Button from '@material-ui/core/Button';
+import Tooltip from '@material-ui/core/Tooltip';
+import { green, red } from '@material-ui/core/colors';
+import { makeStyles } from '@material-ui/styles';
+import EditableTableCell from './EditableTableCell';
+import columnsProps from './columnsProps';
+
+const useStyles = makeStyles(theme => ({
+    button: {
+        maxWidth: theme.spacing(5),
+        minWidth: theme.spacing(5),
+        maxHeight: theme.spacing(3),
+        minHeight: theme.spacing(3),
+        padding: 0
+    },
+    row: {
+        backgroundColor: props => {
+            if (props.toBeSaved) {
+                return green[50];
+            }
+            if (props.toBeDeleted) {
+                return red[50];
+            }
+            return 'transparent';
+        }
+    }
+}));
+
+export default function GroupEditableTableRow({
+    row,
+    columns,
+    editable,
+    updateRow,
+    validateRow,
+    removeRow,
+    isRowValid,
+    resetRow,
+    deleteRowPreEdit,
+    handleEditClick,
+    setRowToBeSaved,
+    setRowToBeDeleted,
+    removeRowOnDelete,
+    ...rest
+}) {
+    const [rowValid, setRowValid] = useState();
+
+    const isRowValidRef = useRef();
+
+    const classes = useStyles({ toBeSaved: row.toBeSaved, toBeDeleted: row.toBeDeleted });
+
+    useEffect(() => {
+        isRowValidRef.current = isRowValid;
+    }, [isRowValid]);
+
+    useEffect(() => {
+        if (row.editing) {
+            let valid = true;
+
+            if (validateRow) {
+                valid = validateRow(row);
+            }
+
+            columns.forEach(column => {
+                if (column.required === true && !row[column.id]) {
+                    valid = false;
```

Comment:
could  if (column.required === true && ...) just become if (column.required && ...), Or do we need to make sure the value is explicitly true?